plugins/module_utils/vars.py

```diff
@@ -0,0 +1,260 @@
+                 list_as_dict=False, keep_empty=False, remove_null=False):
+        # type: (Mapping|Sequence, Mapping|Sequence, bool, bool, bool, bool) -> None
+        """
+        Starting from two similar data structure, a `base`  that acting as the
+        source of data and `changes` that acting to add, update or remove
+        keys/items from the `base`, creates a new one similar strucutre.
+
+        `base` and `changes` are two structure that can be assimilable to
+        `Mapping` and `Sequence`, excluding string. Both must have similar
+        type, if one is a `Mapping` the other too, and if one is a `Sequence`
+        the other too.
+
+        This is originally designed to easily update configurations structures
+        provided by format like JSON, YAML or TOML by ensuring the presence or
+        the absence of some values in keys/items to the result.
+
+        To create the new structure starting from `base` and `changes` iterates
+        over all keys/items in `base` and for each which is also exists in
+        `changes` for a same path in the structure, compares their values. When
+        the compared values in both side are assimilable to dictionaries,
+        operate recursively on them.
+
+        `base` data structure acts as basis for the new one. Keys/items that
+        composing it be present or absent to the result depending on the
+        `present` parameter and to their path in the structure.
+
+        `changes` data structure acts to adding/updating or removing keys/items
+        from the original data. Keys/items that composing it be present or
+        absent to the result depending on the state of `present` parameter and
+        to their path in the structure.
+
+        `present` parameter defines if keys/items in `changes` must be present
+        or not to the result depending on their path in the structure.
+        - When `True`, keys/items in `changes` are used to update or add
+          keys/items at `base`.
+        - When `False`, keys/items in `changes` that for a same path in the
+          structure, also exists in `base` and have same value in both not be
+          present in the result (acts as they are removed from `base`).
+
+        Examples when `present=True` :
+        * with this `base` data :
+          `{'A': {'AA': '1'}, 'B': ['2', '3'], 'C': '4'}`
+        * and this `changes` data :
+          `{'A': {'AB': '9'}, 'B': ['8', '2'], 'C': '7'}`
+        * get this result :
+          `{'A': {'AA': '1', 'AB': '9'}, 'B': ['2', '3', '8'], 'C': '7'}`
+
+        Example when `present=False` :
+        * with this `base` data :
+          `{'A': {'AA': '1', 'AB': '2'}, 'B': ['3', '4'], 'C': '5'}`
+        * and this `changes` data :
+          `{'A': {'AA': '1'}, 'B': ['3', '9'], 'C': '8'}`
+        * get this result :
+          `{'A': {'AB': '2'}, 'B': ['4'], 'C': '5'}`
+
+        `list_as_dict` parameter defines the way by update list items in `base`
```

Comment:
`list_as_dict` sounds very confusing and counter-intuitive given the fact that lists and dicts are intrinsically different.

Reply:
I have found this handy because in fact it converts list in dict and treat it as it. I renamed it into `merge_list_by_index` because when it enabled items are merged by comparing their value for each index position in both side.